apps/designer/app/routes/rest/pages.$projectId.ts

```diff
@@ -1,30 +1,84 @@
-import type { LoaderFunction, ActionFunction } from "@remix-run/node";
+import { LoaderFunction, ActionFunction } from "@remix-run/node";
 import { db } from "@webstudio-is/project/index.server";
-import { Pages } from "@webstudio-is/project";
+import { type Pages, utils } from "@webstudio-is/project";
+import { zfd } from "zod-form-data";
+import { z } from "zod";
+import { PutPageData } from "~/designer/features/sidebar-left/panels/pages/settings";
+import { sentryException } from "~/shared/sentry";
 
-export const action: ActionFunction = async ({ request, params }) => {
-  if (params.projectId === undefined) {
-    throw new Error(`Project ID required`);
-  }
+const nonEmptyString = (fieldName: string) => {
+  const message = `${fieldName} can't be empty`;
+  return z
+    .string({ invalid_type_error: message, required_error: message })
+    .transform((v) => v.trim())
+    .refine((val) => val !== "", message);
+};
+
+const createPageInputSchema = zfd.formData({
```

Comment:
1. Schema name is always capitalized because it corresponds to the type. 
2. I started using Schema suffix and then regretted it, because when not having it we cal literally have the same name for both schema and the type without conflicts, its actually what they use as a convention in zod

```
const Dog = z.object({
  name: z.string(),
  age: z.number(),
});

// extract the inferred type like this
type Dog = z.infer<typeof Dog>;
```


```suggestion
const CreatePageInput = zfd.formData({
```

Reply:
I think keeping `Schema` suffix can be useful in cases when you want to export both the type and the schema

Reply:
that's a thing I didn't know when I used the suffix, you can apparently export both and then when importing you can say if you want the schema or the type using the type 

import {type Test} from '...' // imports type
import {Test} from '...' // imports schema

Reply:
oh wait, you are saing `import { type Test, Test } from "...";` is impossible???

Reply:
Yeah, TypeScript doesn't allow you to import both under the same name.

Reply:
oh you don't need to import both, if you import just the schema namespace you can use it as a type

Reply:
So basically type declaration on the same namespace creates an alias on that namepsace and so it becomes both type and code and can be used for both

Reply:
Yeah, it works. Cool!

```
import { Test } from "./pages.$projectId";
const _test: Test = Test.parse("test");
```